### PR TITLE
Per 9426 add update share link

### PR DESCRIPTION
**Pull request by liam-lloyd**
Compare to per-9424_add_generate_share_link to see changes unique to this PR, if that branch isn't merged yet.

### CODE REVIEW

**Review by cecilia-donnelly — COMMENTED** [review]
Thanks, @liam-lloyd ! Can you add some documentation about how to call this?

Note that in our group review session Liam showed that the best way to test these is to add a new file in the package and call the new function.

File: docs/API.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+#### updateShareLink
+```
+updateShareLink(
+  configuration: ClientConfiguration,
+  shareLink: ShareLink,
+): Promise<ShareLink>
+```
+
+Each non-readonly property of the share link will be updated to match `shareLink`.
```
Comment (cecilia-donnelly) [inline docs/API.md:73]: Is there still that "gotcha" where creating a share link that already exists returns an unchanged link? I know we talked about it but I don't remember what the verdict was.

Comment (liam-lloyd) [reply docs/API.md:75]: It does exist for the v2 endpoint; I've added a note about it

**Review by cecilia-donnelly — APPROVED** [review]
Thanks @liam-lloyd !

**Pull request merged**